tools/lsp-wake/astree.cpp

```diff
@@ -437,10 +482,28 @@ std::string ASTree::sanitizeComment(std::string comment) {
   return comment;
 }
 
+std::string ASTree::composeOuterComment(std::vector<std::pair<std::string, int>> comment) {
+  std::string composed;
+  for (int i = (int) comment.size() - 2; i >= 0; i--) {
+    comment[i].first = sanitizeComment(comment[i].first);
+    composed += comment[i].first + "\n";
+  }
+  return composed;
+}
+
+void ASTree::emplaceComment(std::vector<std::pair<std::string, int>> &comment, const std::string &text, int level) {
+  if (comment.empty() || comment.back().second < level) {
+    comment.emplace_back(text, level);
+  }
+  comment.back().first += text;
+}
+
 ASTree::SymbolUsage::SymbolUsage(Location _usage, Location _definition) : usage(std::move(_usage)), definition(std::move(_definition)) {}
 
-ASTree::Comment::Comment(std::string _comment_text, Location _location)
-: comment_text(std::move(_comment_text)), location(std::move(_location)) {}
+ASTree::Comment::Comment(std::string _comment_text, Location _location, int _level)
+: comment_text(std::move(_comment_text)), location(std::move(_location)) {
+  level = _level;
```

Comment:
You can (and probably should) just add a `level(_level)`

Reply:
Done